colink/protocol.py

```diff
@@ -75,9 +76,10 @@ def run(
         q = queue.Queue()
         for protocol_and_role in self.mapping.keys():  # insert user func to map
             user_func = self.mapping[protocol_and_role]
+            _cl = deepcopy(cl)
             t = threading.Thread(
                 target=thread_func,
-                args=(q, protocol_and_role, cl, vt_public_addr, user_func),
+                args=(q, protocol_and_role, _cl, vt_public_addr, user_func),
```

Comment:
Can you do this in place? Similar to your edit below?

Reply:
Sure.